### PR TITLE
Make the site more responsive to different window sizes

### PR DESCRIPTION
**Pull request by hallgchris**
- On large displays, the sidebar can be shown or hidden, and is shown by default. It does not cover page content.
- On smaller displays, the sidebar again can be shown or hidden, but is hidden by default and when opened, covers page content.
- A button in the top left toggles the sidebar, and on small displays, clicking/tapping outside the sidebar will also close it.
- On large displays, the index page now displays 2 charts on the first row and one on the next, as opposed to 3 on one row.

### CODE REVIEW

**Comment by hallgchris** [comment]
[Better mobile experience on DAS](https://trello.com/c/4ZudBKcj/60-better-mobile-experience-on-das)

File: client/css/common.css [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    padding-bottom: 0px;
+    padding-top: 0px;
+}
+
+/* Sidebar */
+#sidebar {
+    width: inherit;
+    min-width: 220px;
+    max-width: 220px;
+    background-color: #f5f5f5;
```
Comment (khanguslee) [inline client/css/common.css:26]: We should probably create a constants file for these colours. (No need for this PR though)

File: client/css/common.css [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    opacity: 1;
+    float: left;
+    height: 100%;
+    position: relative;
+    overflow-y: auto;
+    overflow-x: hidden;
+    z-index: 10; /* in front of overlay */
```
Comment (khanguslee) [inline client/css/common.css:33]: I would try to steer away from `z-index` (it becomes a hassle later on if all ur elements have a z-index). Is there a reason why you have this? Or is this the *only* way we can put the element in front of the overlay?

File: client/css/common.css [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    margin-left: -250px;
+}
+
+#sidebar-overlay {
+    display: none;
+    position: fixed;
+    width: 100vw;
+    height: 100vh;
+    background: rgba(0, 0, 0, 0.7);
+    z-index: 9; /* behind actual sidebar */
```
Comment (khanguslee) [inline client/css/common.css:52]: See above comment about stance of z-index. (Imagine having to balance 100s of these)

File: client/partials/head.ejs [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<link rel="stylesheet" type="text/css" href="css/bootstrap.css">
```
Comment (khanguslee) [inline client/partials/head.ejs:1]: Good idea! 🎉

File: package.json [hunk 5]
```diff
@@ -35,6 +35,7 @@
     "socket.io": "^2.2.0"
   },
   "devDependencies": {
+    "@fortawesome/fontawesome-free": "^5.9.0",
```
Comment (khanguslee) [inline package.json]: Can you put this as a dependancy?

File: client/js/common.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
+/* eslint-disable no-unused-vars */
```
Comment (khanguslee) [inline client/js/common.js]: IDK how to fix this properly either, but can you use:
```
// eslint-disable-next-line no-unused-vars
```
instead? Easy one liner 😄

**Review by khanguslee — CHANGES_REQUESTED** [review]
Just a couple of small fixes from first glance.

**Comment by khanguslee** [comment]
Just had a quick demo with the test script. Looks amazing on mobile!

**Comment by khanguslee** [comment]
Sidebar gets cut off on every page except for Dashboard and Download Files. pls fix.
image

File: client/page_404.ejs [hunk 7]
```diff
@@ -1,9 +1,8 @@
 <html>
   <head>
+	  <% include ./partials/head %>
```
Comment (harsilspatel) [inline client/page_404.ejs:3]: oh yeah thanks for adding this:)

**Review by harsilspatel — APPROVED** [review]
this looks damn cool on mobile, great work @hallgchris 🤩

**Review by khanguslee — APPROVED** [review]
LGTM

**Pull request merged**